File: check_db.js
const sqlite3 = require('better-sqlite3');
const fs = require('fs');
const path = require('path');

const dbPath = path.join(process.cwd(), '.boukensha', 'events.db');
console.log(`Checking database: ${dbPath}`);

if (!fs.existsSync(dbPath)) {
  console.error('Database not found');
  process.exit(1);
}

try {
  const db = new sqlite3(dbPath);

  // Check event counts by phase
  console.log('\nEvents by phase:');
  const phases = db.prepare(`
    SELECT phase, COUNT(*) as count, COUNT(CASE WHEN tools_sent IS NOT NULL THEN 1 END) as with_tools_sent
    FROM events
    WHERE session_id = '20260803T220918Z-7975a5d3'
    GROUP BY phase
  `).all();

  for (const row of phases) {
    console.log(`  ${row.phase.padEnd(15)} count=${String(row.count).padStart(4)}  with_tools_sent=${String(row.with_tools_sent).padStart(4)}`);
  }

  // Check sample prompt event
  console.log('\nSample prompt event:');
  const prompt = db.prepare(`
    SELECT details FROM events
    WHERE session_id = '20260803T220918Z-7975a5d3' AND phase = 'prompt'
    LIMIT 1
  `).get();

  if (prompt) {
    const event = JSON.parse(prompt.details);
    console.log(`  tool_count: ${event.tool_count}`);
  }

  // Check sample response event
  console.log('\nSample response event:');
  const response = db.prepare(`
    SELECT tools_sent, details FROM events
    WHERE session_id = '20260803T220918Z-7975a5d3' AND phase = 'response'
    LIMIT 1
  `).get();

  if (response) {
    const event = JSON.parse(response.details);
    console.log(`  tools_sent (column): ${response.tools_sent}`);
    console.log(`  tools_sent (json):   ${event.tools_sent}`);
  }

  db.close();
} catch (err) {
  console.error('Error:', err.message);
  process.exit(1);
}

File: check_db.py
#!/usr/bin/env python3
import sqlite3
import json
from pathlib import Path

db_path = Path(".boukensha/events.db")
conn = sqlite3.connect(str(db_path))
conn.row_factory = sqlite3.Row

# Check if tools_sent is populated
cursor = conn.execute("""
SELECT
  phase,
  COUNT(*) as count,
  COUNT(CASE WHEN tools_sent IS NOT NULL THEN 1 END) as with_tools_sent
FROM events
WHERE session_id = '20260803T220918Z-7975a5d3'
GROUP BY phase
""")

print("Events by phase:")
for row in cursor:
    print(f"  {row['phase']:15} count={row['count']:4}  with_tools_sent={row['with_tools_sent']:4}")

# Check a sample prompt event to see tool_count
cursor = conn.execute("""
SELECT details FROM events
WHERE session_id = '20260803T220918Z-7975a5d3'
  AND phase = 'prompt'
LIMIT 1
""")
row = cursor.fetchone()
if row:
    event = json.loads(row['details'])
    print(f"\nSample prompt event tool_count: {event.get('tool_count')}")

# Check a sample response event
cursor = conn.execute("""
SELECT tools_sent, details FROM events
WHERE session_id = '20260803T220918Z-7975a5d3'
  AND phase = 'response'
LIMIT 1
""")
row = cursor.fetchone()
if row:
    event = json.loads(row['details'])
    print(f"Sample response event tools_sent (column): {row['tools_sent']}")
    print(f"Sample response event tools_sent (json):   {event.get('tools_sent')}")

conn.close()
